src/plugins/intel_cpu/src/nodes/multinomial.cpp

```diff
@@ -0,0 +1,287 @@
+    } else {
+        m_samples_count =
+            reinterpret_cast<const int64_t*>(getParentEdgeAt(NUM_SAMPLES_PORT)->getMemoryPtr()->getData())[0];
+    }
+
+    m_batches_count = probs_shape[0];
+    m_probs_count = probs_shape[1];
+    m_samples_probs_count = m_samples_count * m_probs_count;
+    m_input_elements_count = m_batches_count * m_probs_count;
+    m_output_elements_count = m_batches_count * m_samples_count;
+    m_batches_samples_probs_count = m_output_elements_count * m_probs_count;
+}
+
+bool Multinomial::isExecutable() const {
+    return !isInputTensorAtPortEmpty(PROBS_PORT) && !isInputTensorAtPortEmpty(NUM_SAMPLES_PORT);
+}
+
+bool Multinomial::created() const {
+    return getType() == Type::Multinomial;
+}
+
+void Multinomial::execute(dnnl::stream strm) {
+    switch (m_probs_precision) {
+    case ov::element::f32:
+        return execute_probs_type<float>();
+    case ov::element::f16:
+        return execute_probs_type<float16>();
+    case ov::element::bf16:
+        return execute_probs_type<bfloat16_t>();
+    default:
+        THROW_CPU_NODE_ERR("Multinomial CPU implementation does not support probs element type: ", m_probs_precision);
+    }
+}
+
+void Multinomial::executeDynamicImpl(dnnl::stream strm) {
+    execute(strm);
+}
+
+template <typename P>
+void Multinomial::execute_probs_type() {
+    switch (m_output_precision) {
+    case ov::element::i32:
+        return execute_convert_type<P, int32_t>();
+    default:
+        THROW_CPU_NODE_ERR("Multinomial CPU implementation does not support output convert type: ", m_output_precision);
+    }
+}
+
+template <typename P, typename O>
+void Multinomial::execute_convert_type() {
+    const auto* probs = reinterpret_cast<const P*>(getParentEdgeAt(PROBS_PORT)->getMemoryPtr()->getData());
+    auto* output = reinterpret_cast<O*>(getChildEdgeAt(OUTPUT_PORT)->getMemoryPtr()->getData());
+
+    std::vector<P> m_cdf(m_input_elements_count);
+    std::vector<P> m_max_per_batch(m_batches_count);
+    std::vector<P> m_random_samples(m_output_elements_count);
+
+    // exp & cumsum
+    if (m_log_probs) {
+        parallel_for(m_batches_count, [&](size_t idx) {
+            const auto start_idx = idx * m_probs_count;
+            m_cdf[start_idx] = std::exp(probs[start_idx]);
+            for (size_t prev = start_idx, curr = prev + 1; curr < (start_idx + m_probs_count); ++prev, ++curr) {
+                m_cdf[curr] = std::exp(probs[curr]) + m_cdf[prev];
+            }
+        });
+    } else {
+        parallel_for(m_batches_count, [&](size_t idx_batch) {
+            const auto start_idx = idx_batch * m_probs_count;
+            const auto* probs_start_idx = probs + start_idx;
+            std::partial_sum(probs_start_idx, probs_start_idx + m_probs_count, m_cdf.begin() + start_idx);
+        });
+    }
+
+    // TODO RandomUniform - should use RandomUniform kernel to match other frameworks' seed results
```

Comment:
Do we have ticket numbet for that TODO? 
Just wondering when we are going to follow-up that?

Reply:
yes, will send you via PM